### ISSUE TITLE
[Data Issue]: 

### ISSUE THREAD

**Issue by leon-hard** [electricitymaps/electricitymaps-contrib#6738]
### When did this happen?

Last weekend (10th May to 13 th May '24)

### What zones are affected?

DE

### What is the problem?

There seems to be a problem with the estimation models. Currently, there is a large deviation for wind:

EM estimation: 5.33 GW
Sum from TSO values (actual reportings, no estimation): 22.438 GW

Difference is 17.108 GW or factor 4.2.

<img width="948" alt="2024-05-13 07_30_42-Electricity Production _ Energy-Charts – Mozilla Firefox" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/7898710/c945a4c5-0cc6-4258-8bec-7a76ef26c075">

**Comment by VIKTORVAV99** [comment]
We are already aware and internal issue has been created but thanks of opening this.

We only got partial data from ENTSO-E which hit an edge case in our data processing.

Hopefully we should have this resolved soon!

**Comment by leon-hard** [comment]
Removing or altering the color for countries where the estimation error is significant (e.g., a factor greater than 2) could be a way to visually communicate the reliability of the data. I thought this had already been implemented some time ago. For instance, I also observed an estimation error of 14.5 % for solar from 10:00 to 11:00 for 12th of May, which can be significant, as solar was the main source of electricity by then.

I saw plenty of post on social media throughout the weekend which were using EM data based on wrong estimations for Germany. On Sunday, for instance, the estimation of too less solar was compensated be an estimation of gas and coal which was also off by factors > 2.

**Comment by VIKTORVAV99** [comment]
> Removing or altering the color for countries where the estimation error is significant (e.g., a factor greater than 2) could be a way to visually communicate the reliability of the data. I thought this had already been implemented some time ago. For instance, I also observed an estimation error of 14.5 % for solar from 10:00 to 11:00 for 12th of May, which can be significant, as solar was the main source of electricity by then.
> 
> 
> 
> I saw plenty of post on social media throughout the weekend which were using EM data based on wrong estimations for Germany. On Sunday, for instance, the estimation of too less solar was compensated be an estimation of gas and coal which was also off by factors > 2.

It was implemented for a few zones that used TSA but we switched to an estimation system that uses forecasts from entso-e combined with our own forecasts. However we wrongly assumed that the forecast would either be complete or completely missing. Not just having partial wind data.

The solar values are what we directly got from the ENTSO-E day ahead values as intraday and total are missing for 1 of the two wind modes and solar.

**Comment by leon-hard** [comment]
I totally agree that it is a pity that the transparency requirements by European legislation are not tight enough regarding data quality/availability (and licensing, but that's another story). 

However, I strongly recommend to adopt a more cautious approach by choosing not to display data when its accuracy cannot be sufficiently guaranteed. This practice is crucial because erroneous data can easily be misused to support specific agendas, particularly by those who do not necessarily prioritize the accuracy or origin of the information.

For instance, data from ENTSO-E, and consequently also from EM, indicate an apparent 31% increase in gas-driven electricity production in Germany from 2022 to 2023. However, this assertion is contradicted by several other authoritative sources, including the German Statistical Office, industry reports from BDEW, the AG Energiebilanzen workgroup, and gas import statistics, all of which suggest a slight decrease in production. This discrepancy underscores a broader, structural issue with ENTSO-E data management practices: historical data errors often remain uncorrected, and the prioritization of data consistency over data quality can lead to significant inaccuracies in reported data trends.

**Comment by leon-hard** [comment]
I have noticed that you had greyed out the country with unriable data. I think this is much better than showing wrong data.

However, I have a brief inquiry about the flow tracing algorithm. I noticed that emission values from countries, where data quality may not be reliable, are still being used in the algorithm. Or am I understanding the tooltips in a wrong way? Can you maybe confirm this?

![grafik](https://github.com/electricitymaps/electricitymaps-contrib/assets/7898710/5b470001-1401-4e15-a9c2-b0c52ba6148d)

**Comment by VIKTORVAV99** [comment]
> I have noticed that you had greyed out the country with unriable data. I think this is much better than showing wrong data.
> 
> However, I have a brief inquiry about the flow tracing algorithm. I noticed that emission values from countries, where data quality may not be reliable, are still being used in the algorithm. Or am I understanding the tooltips in a wrong way? Can you maybe confirm this?
> 
> ![grafik](https://private-user-images.githubusercontent.com/7898710/330068950-5b470001-1401-4e15-a9c2-b0c52ba6148d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2MTAyNjYsIm5iZiI6MTcxNTYwOTk2NiwicGF0aCI6Ii83ODk4NzEwLzMzMDA2ODk1MC01YjQ3MDAwMS0xNDAxLTRlMTUtYTljMi1iMGM1MmJhNjE0OGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTNUMTQxOTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmUxMzU0YTUzOTc4YTAxY2E3ZjUxN2ZlYTc2NjJhOGE5NThkOTA5MzAzYjY0MjljMDFjOTRmYjhlNjc5YzBlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EjOvrJwh9nOctjEdT3HI9ldWwFD76rMdAeXLhlJe48Q)

It is still being used in the flow tracing algorithm for the simple reason that even if the data is unreliable right now it is still more accurate than a static value would be and if we don't include this data the neighbouring countries would have a higher degree of error than if we used it.

In this situation we are choosing to use the best of two bad solutions unfortunately.

**Comment by leon-hard** [comment]
Thank you very much for your swift response!

I would argue that it is better not to use data if you know that it is unreliable. I would suggest to admit that part of the import is unknown and not just show any data. I don't see why this could not be implemented. But I clearly understand that it is always a compromise between data availability and quality.

Best would be for sure to just get proper input data ;-)

**Comment by VIKTORVAV99** [comment]
> Thank you very much for your swift response!
> 
> I would argue that it is better not to use data if you know that it is unreliable. I would suggest to admit that part of the import is unknown and not just show any data. I don't see why this could not be implemented. But I clearly understand that it is always a compromise between data availability and quality.
> 
> Best would be for sure to just get proper input data ;-)

We could say it is unknown but if someone ran the numbers it would not add up. The flowtracing system needs a co2 intensity value if there is a flow values.

Getting proper input data is harder than it should be 😅

**Comment by VIKTORVAV99** [comment]
This should be resolved now for the ENTSO-E parser and the missing data has been corrected.
